flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/job/coordination/ClientCoordinationHandlerTest.java

```diff
@@ -0,0 +1,93 @@
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.rest.handler.HandlerRequest;
+import org.apache.flink.runtime.rest.handler.RestHandlerException;
+import org.apache.flink.runtime.rest.messages.job.coordination.ClientCoordinationHeaders;
+import org.apache.flink.runtime.rest.messages.job.coordination.ClientCoordinationMessageParameters;
+import org.apache.flink.runtime.rest.messages.job.coordination.ClientCoordinationRequestBody;
+import org.apache.flink.runtime.webmonitor.TestingRestfulGateway;
+import org.apache.flink.util.SerializedValue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/** Tests for {@link ClientCoordinationHandler}. */
+public class ClientCoordinationHandlerTest {
+    private static final JobID TEST_JOB_ID = new JobID();
+    private ClientCoordinationHandler clientCoordinationHandler;
+    private HandlerRequest<ClientCoordinationRequestBody> testRequest;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        final TestingRestfulGateway testingRestfulGateway =
+                new TestingRestfulGateway.Builder().build();
+
+        clientCoordinationHandler =
+                new ClientCoordinationHandler(
+                        () -> CompletableFuture.completedFuture(testingRestfulGateway),
+                        Time.seconds(10),
+                        Collections.emptyMap(),
+                        ClientCoordinationHeaders.getInstance());
+
+        Map<String, String> pathParameters = new HashMap<>();
+        pathParameters.put("jobid", TEST_JOB_ID.toString());
+        pathParameters.put(
+                "operatorid", OperatorID.fromJobVertexID(new JobVertexID()).toHexString());
+        testRequest =
+                HandlerRequest.resolveParametersAndCreate(
+                        new ClientCoordinationRequestBody(
+                                new SerializedValue<>(new TestClientCoordinationRequest())),
+                        new ClientCoordinationMessageParameters(),
+                        pathParameters,
+                        Collections.emptyMap(),
+                        Collections.emptyList());
+    }
+
+    @Test
+    public void testGetResponseException() {
+        final TestingRestfulGateway testingRestfulGateway =
+                new TestingRestfulGateway.Builder()
+                        .setDeliverCoordinationRequestToCoordinatorFunction(
+                                (jobId, operatorId, request) -> {
+                                    throw new RuntimeException("Throw runtime exception");
+                                })
```

Comment:
return a failed future instead; you're testing behavior that doesnt occur at runtime.